### PR TITLE
Update of the docs

### PR DESCRIPTION
**Pull request by kang-makes**
With the implementation of the common library, a couple of values changed and some features have been added.

We can also add more documentation to this chart.

## Note to the reviewer

There is a spurious change in the `templates/kubelet/daemonset.yaml` file. As I have to bump the version to change the documentation I am fixing also that bug that I found while I was doing the docs.

### CODE REVIEW

File: charts/newrelic-infrastructure/README.md [hunk 1]
```diff
@@ -1,29 +1,102 @@
...
+  config:
+    interval: 15s
+
+lowDataMode: false
+```
+
+The `lowDataMode` toggle is the simplest way amd setting it to `true` changes the default scrape interval from 15 seconds (the default) to 30 seconds.
+
+If you need for some reason to fine tune the amount of seconds you can use `common.config.interval` directly. If you take a look to the `values.yaml`
+file, the value there is `nil`. If set any value there, the `lowDataMode` toggle is ignored and this value is issued.
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:56]: "issued"? I am not sure it is correct 🤔 "used" instead?

File: charts/newrelic-infrastructure/README.md [hunk 2]
```diff
@@ -1,29 +1,102 @@
...
+
+Setting this interval above 40 seconds can make you experience issues with the Kubernetes Cluster Explorer so this chart limits setting the interval
+inside the range of 10 to 40 seconds.
+
+### Affinities and tolerations
+
+The New Relic common library allows to set affinities, tolerations and nodeSelectors in globally with the resto of the integrations and products. This
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:63]:
```suggestion
The New Relic common library allows to set affinities, tolerations and nodeSelectors globally with the rest of the integrations and products. This
```

File: charts/newrelic-infrastructure/README.md [hunk 3]
```diff
@@ -1,29 +1,102 @@
...
+integration in particular has affinities and tolerations set to be able to schedule pods in nodes that are tainted as master nodes and to schedule a
+pod near the KSM to reduce the inter-node traffic.
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:65]:
```suggestion
integration in particular has affinities and tolerations set to be able to schedule kubelet scraper components on nodes that are tainted as master nodes and
to schedule the KSM scraper on the same node of KSM to reduce the inter-node traffic.
```

File: charts/newrelic-infrastructure/README.md [hunk 4]
```diff
@@ -1,29 +1,102 @@
...
+
+Take a look to the [`values.yaml`](values.yaml) so see how to configure them if you are having problems scheduling pods where you want to.
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:67]: I do not like that much this sentence, can we improve it somehow?

File: charts/newrelic-infrastructure/README.md [hunk 5]
```diff
@@ -1,29 +1,102 @@
...
+
+### `hostNetwork` toggle
+
+In versions below v3, changing the `privileged` mode affected the `hostNetwork`. We changed this behavior and now you can set pods to use `hostNetwork`
+using the corresponding [flags from the common library](https://github.com/newrelic/helm-charts/blob/master/library/common-library/README.md)
+(`.global.hostNetwork` and `.hostNetwork`) but the component that scrapers data from the control plane has always set `hostNetwork` to true (Look in the
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:73]:
```suggestion
(`.global.hostNetwork` and `.hostNetwork`) but the component that scrapes data from the control plane has `hostNetwork` enabled by default (Look in the
```

File: charts/newrelic-infrastructure/README.md [hunk 6]
```diff
@@ -1,29 +1,102 @@
...
+If your cluster security policy does not allow to use `hostNetwork`, you can disable it control plane monitoring by setting `controlPlane.enabled` to
+`false.`
+
+### `privileged` toggle
+
+The default value for `privileged` [from the common library](https://github.com/newrelic/helm-charts/blob/master/library/common-library/README.md) is
+`false`.
+
+In this chart it is set to `true` (Look in the [`values.yaml`](values.yaml) for `privileged: true`) because it set `kubelet` pods (the ones that scrape
+metrics from the hosts itself) into privileged mode so it can fetch more fine-grained cpu, memory, process and network metrics for your nodes.
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:87]: I would rephrase also this a bit if possible

File: charts/newrelic-infrastructure/README.md [hunk 7]
```diff
@@ -1,29 +1,102 @@
...
+
+If your cluster security policy does not allow to to have `privileged` in your pod' security context, you can disable it by setting `privileged` to
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:89]: could we tell customer about possible drawbacks? I think there is a NR page somewhere saying what we are losing without privileged=true

File: charts/newrelic-infrastructure/README.md [hunk 8]
```diff
@@ -1,29 +1,102 @@
...
 
 ## Values
 
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | common | object | See `values.yaml` | Config that applies to all instances of the solution: kubelet, ksm, control plane and sidecars. |
 | common.agentConfig | object | `{}` | Config for the Infrastructure agent. Will be used by the forwarder sidecars and the agent running integrations. See: https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/configuration/infrastructure-agent-configuration-settings/ |
-| common.config.interval | duration | `15s` if `lowDataMode == false`, `30s` otherwise. | Intervals larger than 40s are not supported and will cause the NR UI to not behave properly. Any non-nil value will override the `lowDataMode` default. |
+| common.config.interval | duration | `15s` (See [Low data mode](README.md#low-data-mode)) | Intervals larger than 40s are not supported and will cause the NR UI to not behave properly. Any non-nil value will override the `lowDataMode` default. |
+| containerSecurityContext | string | `nil` | This is still to be decided in a follow-up PR regarding privileged mode and defaults changing in an evil way |
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:99]: Shouldn't we decide this before publishing? 

Anyway, I would not put that into the readme, but in a issue.

Being a user I would be scared reading that

File: charts/newrelic-infrastructure/README.md (hunk 7 above)
Comment (paologallinaharbur) [reply charts/newrelic-infrastructure/README.md:89]: I am sure it was somewhere, but I am not able to find it

Comment (paologallinaharbur) [reply charts/newrelic-infrastructure/README.md:89]: not sure it is only this https://docs.newrelic.com/docs/infrastructure/infrastructure-monitoring/infrastructure-security/infrastructure-security#linux-unprivileged

File: charts/newrelic-infrastructure/README.md (hunk 1 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md:56]: I don't know what I was thinking. I have to install something to check the grammar from the IDE because I have you doing this ungrateful work...

File: charts/newrelic-infrastructure/README.md (hunk 8 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md:99]: It was a leftover. This is completely defined in the previous PR. This task was also meant to change this but I overlooked it.

File: charts/newrelic-infrastructure/README.md [hunk 9]
```diff
@@ -1,29 +1,102 @@
...
+There are two mechanisms to reduce the amount of data that this integration sends to New Relic. See this snippet from the `values.yaml` file:
+```yaml
+common:
+  config:
+    interval: 15s
+
+lowDataMode: false
+```
+
+The `lowDataMode` toggle is the simplest way amd setting it to `true` changes the default scrape interval from 15 seconds (the default) to 30 seconds.
```
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: I rewrite a part of it in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

File: charts/newrelic-infrastructure/README.md (hunk 1 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: I rewrite a part of it in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

File: charts/newrelic-infrastructure/README.md (hunk 3 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: Solved in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

File: charts/newrelic-infrastructure/README.md (hunk 4 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: I rewrite a part of it in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

**Comment by paologallinaharbur** [comment]
I added the skip-e2e labels just to save actions minutes, if you need them remove the label :)

File: charts/newrelic-infrastructure/README.md (hunk 5 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: Solved in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

File: charts/newrelic-infrastructure/README.md (hunk 6 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: I rewrite it in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

File: charts/newrelic-infrastructure/README.md (hunk 7 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: I added more info in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

File: charts/newrelic-infrastructure/README.md (hunk 8 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: Solved in [#0fb8df2](https://github.com/newrelic/nri-kubernetes/pull/409/commits/0fb8df26c7f339111945dbe2bd0c2af63943cbc0)

**Comment by kang-makes** [comment]
As it was the final tests I leaved it to be completely sure that is ready to release.

But I am only changing the README so it is OK.

File: charts/newrelic-infrastructure/README.md [hunk 10]
```diff
@@ -1,29 +1,119 @@
...
+If you need for some reason to fine-tune the number of seconds you can use `common.config.interval` directly. If you take a look at the `values.yaml`
+file, the value there is `nil`. If any value is set there, the `lowDataMode` toggle is ignored as this value takes precedence.
+
+Setting this interval above 40 seconds can make you experience issues with the Kubernetes Cluster Explorer so this chart limits setting the interval
+inside the range of 10 to 40 seconds.
+
+### Affinities and tolerations
+
+The New Relic common library allows to set affinities, tolerations, and node selectors globally with the rest of the integrations and products but this
+chart has an extra level of granularity to set them.
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:57]:
```suggestion
When leveraged in a chart, the New Relic common library allows to set affinities, tolerations, and node selectors globally. However, this
chart has an extra level of granularity that allows to configure them for each ones of the components.
```

I am not sure about this, but the old sentence was quite confusing

File: charts/newrelic-infrastructure/README.md [hunk 11]
```diff
@@ -1,29 +1,119 @@
...
+
+kubelet:
+  affinity: {}
+ksm:
+  affinity: {}
+controlPlane:
+  affinity: {}
+```
+
+The order to set an affinity is to set first any `kubelet.affinity`, `ksm.affinity`, or `controlPlane.affinity`. If they are empty it fallbacks to
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:74]:
```suggestion
Rendering a component affinity, the chart gives precedence to any affinity specified at component level: `kubelet.affinity`, `ksm.affinity`, or `controlPlane.affinity`. If they are empty it fallbacks to
```

To me what is confusing is the "it" not referring to anything in the second sentence. For that reason I added "the chart" in the first sentence.
(but do not trust at all my grammar skills 😄 )

File: charts/newrelic-infrastructure/README.md (hunk 10 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: Solved in [#3d77275](https://github.com/newrelic/nri-kubernetes/pull/409/commits/3d77275993abcbfdd3bff2660bc3b64e18abf656)

File: charts/newrelic-infrastructure/README.md [hunk 12]
```diff
@@ -1,29 +1,120 @@
...
+If your cluster security policy does not allow to use `hostNetwork`, you can disable it control plane monitoring by setting `controlPlane.enabled` to
+`false.`
+
+### `privileged` toggle
+
+The default value for `privileged` [from the common library](https://github.com/newrelic/helm-charts/blob/master/library/common-library/README.md) is
+`false` but in this particular this chart it is set to `true` (Look in the [`values.yaml`](values.yaml) for `privileged: true`)
 
-## Requirements
+This is because when `kubelet` pods need to run in privileged mode to fetch cpu, memory, process, and network metrics for your nodes.
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/README.md:102]:
```suggestion
This is because `kubelet` pods need to run in privileged mode to fetch cpu, memory, process, and network metrics of your nodes.
```

**Review by paologallinaharbur — APPROVED** [review]

File: charts/newrelic-infrastructure/README.md (hunk 12 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: Solved in [#85e36f5](https://github.com/newrelic/nri-kubernetes/pull/409/commits/85e36f5115754dce005f8970e931c32630797de5)

File: charts/newrelic-infrastructure/README.md (hunk 11 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: Solved in [#85e36f5](https://github.com/newrelic/nri-kubernetes/pull/409/commits/85e36f5115754dce005f8970e931c32630797de5)

File: charts/newrelic-infrastructure/README.md (hunk 10 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/README.md]: Solved in [#85e36f5](https://github.com/newrelic/nri-kubernetes/pull/409/commits/85e36f5115754dce005f8970e931c32630797de5)
